providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/Operator.java

```diff
@@ -53,15 +59,18 @@ static class FlagProperties {
             if (from instanceof Map) {
                 Map<?, ?> dataMap = (Map<?, ?>) from;
 
-                Object flagKey = dataMap.get(FLAG_KEY);
+                final Object flagKey = Optional.ofNullable(dataMap.get(FLAGD_PROPS_KEY))
```

Comment:
@Kavindu-Dodan there's not a lot of Optional usage in this repo, but the alternative was really ugly and not immutable.

Reply:
This looks fine :)